include/coap2/coap_mutex.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * coap_mutex.h -- mutex utilities
+ *
+ * Copyright (C) 2019 Jon Shallow <supjps-libcoap@jpshallow.com>
+ *
+ * This file is part of the CoAP library libcoap. Please see README for terms
+ * of use.
+ */
+
+/**
+ * @file coap_mutex.h
+ * @brief COAP mutex mechanism wrapper
+ */
+
+#ifndef _COAP_MUTEX_H_
```

Comment:
IIRC we have previously agreed to not use the leading underscore anymore to avoid confusion with OS-specific symbols.

Reply:
Oops - I had just copied the file across form tinydtls and changed dtls to coap as appropriate without thinking.  Now fixed in libcoap on the next push.